### PR TITLE
feat: add i18n_js_namespace to support atlas - FC-0012

### PR DESCRIPTION
**Pull request by shadinaif**
Implement OEP-58 JavaScript translations
---

Details:
* Proposal: https://github.com/openedx/edx-platform/pull/33166
* Supporting platform pull request: https://github.com/openedx/edx-platform/pull/33698
* Similar pull request: https://github.com/openedx/xblock-drag-and-drop-v2/pull/365

Testing
---

Local tutor 17.0.2
- [x] translations are still working including JS translations. I added a temporary console log and it showed the translation correctly (to avoid the hustle connecting a real LTI)
`console.log(gettext('LTI Consumer'))`
Screenshot from 2024-02-28 11-58-10
 
References
---

This pull request is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which implements the [Translation Infrastructure update OEP-58](https://docs.openedx.org/en/latest/developers/concepts/oep58.html).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by shadinaif** [comment]
please review @OmarIthawi

File: lti_consumer/lti_xblock.py [hunk 1]
```diff
@@ -776,6 +794,15 @@ def get_pii_sharing_enabled(self):
...
         return True
 
+    @property
+    def i18n_service(self):
+        """ Obtains translation service """
+        i18n_service = self.runtime.service(self, "i18n")
+        if i18n_service:
+            return i18n_service
+        else:
+            return DummyTranslationService()
```
Comment (OmarIthawi) [inline lti_consumer/lti_xblock.py:804]: Please revert this change. I don't think it's needed.

File: lti_consumer/lti_xblock.py [hunk 2]
```diff
@@ -678,6 +683,19 @@ def _get_statici18n_js_url(loader):  # pragma: no cover
                 return text_js.format(lang_code=code)
         return None
 
+    def _get_statici18n_js_url(self, loader):
+        """
+        Return the JavaScript translation file provided by the XBlockI18NService.
+        """
+        if url_getter_func := getattr(self.i18n_service, 'get_javascript_i18n_catalog_url', None):
```
Comment (OmarIthawi) [inline lti_consumer/lti_xblock.py:690]:
```suggestion
        if i18n_service := self.runtime.service(self, "i18n"):
            if url_getter_func := getattr(i18n_service, 'get_javascript_i18n_catalog_url', None):
```

No need for the `DummyTranslationService` imho. Keeping the code simpler is better.

**Review by OmarIthawi — COMMENTED** [review]
Thanks @shadinaif!!

Please remove non-essential mocking code. That should make the PR easier to review.

**Comment by OmarIthawi** [comment]
@shadinaif please mark this one as draft!

**Comment by OmarIthawi** [comment]
being merged in https://github.com/openedx/xblock-lti-consumer/pull/445

**Comment by openedx-webhooks** [comment]
@shadinaif Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-lti-consumer&entry.867055334=https://github.com/openedx/xblock-lti-consumer/pull/452&entry.1484655318=https://github.com/shadinaif&entry.752974735=2024-02-28+09:18&entry.1917517419=2024-03-17+15:00&entry.2133058324=No) so we can improve your experience in the future.
